### PR TITLE
Web client migration

### PR DESCRIPTION
**Pull request by janinedotgm**
preview: https://deploy-preview-19--developer-center.netlify.app/migration/migration-web-developers

### CODE REVIEW

File: migration/migration-web-developers.md [hunk 1]
```diff
@@ -0,0 +1,251 @@
+# Migration Guide for Web Developers
+
+With our shift from Proof of Work (PoW) to Proof of Stake (PoS), significant updates have been made to the Nimiq Web Client. This comparison page is meticulously crafted to guide you through the enhancements implemented in the new version. We'll highlight critical changes in configuration, client instantiation, wallet creation, transaction handling, and more.
```
Comment (onmax) [inline migration/migration-web-developers.md]: Maybe add link?

```suggestion
With our shift from Proof of Work (PoW) to Proof of Stake (PoS), significant updates have been made to the [Nimiq Web Client](/build/web-client/). This comparison page is meticulously crafted to guide you through the enhancements implemented in the new version. We'll highlight critical changes in configuration, client instantiation, wallet creation, transaction handling, and more.
```

File: migration/migration-web-developers.md [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+```JavaScript
+const privateKey = Nimiq.PrivateKey.fromHex(privateKeyHex);
+const keyPair = Nimiq.KeyPair.derive(privateKey);
+
+const address = keyPair.toAddress();
+```
+
+### Load a wallet from an encrypted private key
+Previously, you could use the ```Wallet.loadEncrypted()``` function to derive a wallet from an encrypted private key. This function is no longer available.
```
Comment (onmax) [inline migration/migration-web-developers.md]:
```suggestion
Previously, you could use the `Wallet.loadEncrypted()` function to derive a wallet from an encrypted private key. This function is no longer available.
```

File: migration/migration-web-developers.md [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+const wallet = await Wallet.loadEncrypted(encryptedHex, password);
+```
+
+#### Now (PoS)
+
+```JavaScript
+// No equivalent yet
+```
+
+### Export the key pair or private key as a plain Uint8Array
```
Comment (onmax) [inline migration/migration-web-developers.md]:
```suggestion
### Export the key pair or private key as a plain `Uint8Array`
```

File: migration/migration-web-developers.md [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+To export the key pair or private key as a plain Uint8Array, you need to serialize the key pair or private key.
```
Comment (onmax) [inline migration/migration-web-developers.md]:
```suggestion
To export the key pair or private key as a plain `Uint8Array`, you need to serialize the key pair or private key.
```

File: migration/migration-web-developers.md [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+
+#### Now (PoS)
+
+```JavaScript
+const publickeyHex = keyPair.publicKey.toHex();
+const privatekeyHex = keyPair.privateKey.toHex();
+```
+
+## Nimiq.Client
+The Client class is a central component of the Nimiq's new web client framework and is essential for interacting with the Nimiq blockchain network under the PoS consensus model.
```
Comment (onmax) [inline migration/migration-web-developers.md]:
```suggestion
The Client class is a central component of the Nimiq's new Web Client framework and is essential for interacting with the Nimiq blockchain network under the PoS consensus model.
```

File: migration/migration-web-developers.md [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+const client = configBuilder.instantiateClient();
+```
+
+#### Now (PoS)
+
+When using the /web package export, you need to manually call the init function:
+
+```JavaScript
+// When using the /web package export
+import init, * as Nimiq from '@nimiq/core-web/web';
```
Comment (onmax) [inline migration/migration-web-developers.md:167]: In the next version `@nimiq/core-web` will be `@nimiq/core`

File: migration/migration-web-developers.md [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+
+await Nimiq.init();
+Nimiq.GenesisConfig.test(); // Select testnet
+const configBuilder = Nimiq.Client.Configuration.builder();
+const client = configBuilder.instantiateClient();
+```
+
+#### Now (PoS)
+
+When using the /web package export, you need to manually call the init function:
```
Comment (onmax) [inline migration/migration-web-developers.md:163]: There are slight differences between webpack and vite. What about adding a link to the integrations?

File: migration/migration-web-developers.md [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+- ```public getTransactionReceipt(hash: Hash | string)```
+- ```public getTransactionReceiptsByHashes(hashes: Array<Hash | string>)```
+- ```public addBlockListener(listener: BlockListener)```
+
+## Nimiq.Account
+
+In addition to the existing account types ```basic (0)```, ```vesting (1)```, and ```htlc (2)```, Nimiq now has a fourth account type ```staking (3)```. Only the staking contract can be type ```staking```, no other account can have this type.
+
+### User-friendly addresses
+The new library doesn’t yet support the withSpaces parameter. User-friendly addresses are always returned with spaces.
```
Comment (onmax) [inline migration/migration-web-developers.md]:
```suggestion
The new library doesn’t yet support the `withSpaces` parameter. User-friendly addresses are always returned with spaces.
```

File: migration/migration-web-developers.md [hunk 9]
```diff
@@ -0,0 +1,251 @@
...
+```JavaScript
+function lunasToCoins(lunas: number): number {
+  return lunas / 1e5;
+}
+```
+
+## Other
+
+### Send transactions
+**New capability:** Can now handle UintArray
```
Comment (onmax) [inline migration/migration-web-developers.md]:
```suggestion
**New capability:** Can now handle `UintArray`
```

File: migration/migration-web-developers.md [hunk 10]
```diff
@@ -0,0 +1,251 @@
...
+
+### Transaction state
+```MINED``` is replace by ```INCLUDED```
```
Comment (onmax) [inline migration/migration-web-developers.md]:
```suggestion
`MINED` is replace by `INCLUDED`
```

File: migration/migration-web-developers.md [hunk 11]
```diff
@@ -0,0 +1,251 @@
...
+
+## Timestamps
+Timestamps of transactions and blocks are now in milliseconds, where they were in seconds (UNIX) before. That means you no longer have to multiply them by 1000 to use with new Date() in Javascript:
```
Comment (onmax) [inline migration/migration-web-developers.md]:
```suggestion
Timestamps of transactions and blocks are now in milliseconds, where they were in seconds (UNIX) before. That means you no longer have to multiply them by 1000 to use with `new Date()` in JavaScript:
```

**Review by onmax — APPROVED** [review]
Apart from the small comments I left looks good 👍 

--- 

Maybe you already know this, but maybe we could use the [code group component](https://vitepress.dev/guide/markdown#code-groups) from vitepress.

Good thing is less scrolling. Bad thing, some information is hidden and the user has to click.

It would look like this.
image

Up to you :), you can also check what other components we have available for the future

File: migration/migration-web-developers.md [hunk 12]
```diff
@@ -0,0 +1,251 @@
+# Migration Guide for Web Developers
+
+With our shift from Proof of Work (PoW) to Proof of Stake (PoS), significant updates have been made to the [Nimiq Web Client](/build/web-client/). This comparison page is meticulously crafted to guide you through the enhancements implemented in the new version. We'll highlight critical changes in configuration, client instantiation, wallet creation, transaction handling, and more.
```
Comment (ines-toupeira) [inline migration/migration-web-developers.md]:
```suggestion
With our shift from Proof-of-Work (PoW) to Proof-of-Stake (PoS), significant updates have been made to the [Nimiq Web Client](/build/web-client/). This comparison page is meticulously crafted to guide you through the enhancements implemented in the new version. We'll highlight critical changes in configuration, client instantiation, wallet creation, transaction handling, and more.
```

Comment (ines-toupeira) [inline migration/migration-web-developers.md]:
```suggestion
With our shift from Proof of Work (PoW) to Proof of Stake (PoS), the [Nimiq Web Client](/build/web-client/) has undergone significant updates. This comparison page is meticulously crafted to guide you through the enhancements implemented in the new version. We'll highlight critical changes in configuration, client instantiation, wallet creation, transaction handling, and more.
```

File: migration/migration-web-developers.md [hunk 13]
```diff
@@ -0,0 +1,251 @@
...
+#### Now (PoS)
+You now have to manually divide any amount in luna by 10.000 to get NIM:
+
+```JavaScript
+function lunasToCoins(lunas: number): number {
+  return lunas / 1e5;
+}
+```
+
+## Other
```
Comment (ines-toupeira) [inline migration/migration-web-developers.md]: suggestion: maybe move the "Other" to the end? after the "Timestamps"

**Review by ines-toupeira — APPROVED** [review]

**Review by ines-toupeira — APPROVED** [review]

**Pull request merged**